eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -1,3 +1,34 @@
+# +----+------------------+---------+---------------+-----------+---------+------------+-------+---------+---------------+--------+
+# | ## | OS/arch/compiler | Release | Preconditions | Map files | Logging | Unit tests | Mocks | Samples | Code coverage | Linter |
+# +----+------------------+---------+---------------+-----------+---------+------------+-------+---------+---------------+--------+
+# |  1 | Linux ARM        |    +    |               |     +     |    +    |     N/A    |       |   N/A   |               |        |
+# |  2 | Linux ARM        |         |               |           |         |     N/A    |       |   N/A   |               |        |
+# |  3 | Linux (x64)      |    +    |               |     +     |    +    |      +     |   +   |         |               |        |
+# |  4 | Linux (x64)      |         |               |           |    +    |      +     |   +   |    +    |               |        |
+# |  5 | Windows x86      |    +    |               |     +     |         |      +     |       |         |               |        |
+# |  6 | MacOS (x64)      |    +    |               |           |    +    |      +     |       |    +    |               |        |
+# |  7 | MacOS (x64)      |         |               |     +     |    +    |      +     |       |    +    |               |        |
+# |  8 | Linux (x64) GCC5 |    +    |               |           |         |      +     |       |         |               |        |
+# |  9 | Linux (x64) GCC5 |         |               |     +     |         |      +     |       |    +    |               |        |
```

Comment:
Looks like non-release generated map files aren't super useful, especially in exchange of release map files on a particular compiler, like GCC5, or Clang (for Mac). We are losing useful info if we turn map files on debug on such platforms and not release.

```suggestion
# |  9 | Linux (x64) GCC5 |         |               |           |         |      +     |       |    +    |               |        |
```